src/creature_instances.c

```diff
@@ -722,10 +724,74 @@ long instf_first_person_do_imp_task(struct Thing *creatng, long *param)
     TRACE_THING(creatng);
     MapSlabCoord slb_x = subtile_slab_fast(creatng->mappos.x.stl.num);
     MapSlabCoord slb_y = subtile_slab_fast(creatng->mappos.y.stl.num);
+    struct SlabMap* slb;
     if (check_place_to_pretty_excluding(creatng, slb_x, slb_y))
     {
         instf_pretty_path(creatng, NULL);
-    } else
+    }
+    else if (first_person_dig_claim_mode)
+    {
+        if ( check_place_to_convert_excluding(creatng, slb_x, slb_y) )
+        {
+            slb = get_slabmap_block(slb_x, slb_y);
+            struct SlabAttr* slbattr = get_slab_attrs(slb);
+            instf_destroy(creatng, NULL);
+            if (slbattr->category == SlbAtCtg_RoomInterior)
+            {
+                struct Room* room = room_get(slb->room_index);
+                if (!room_is_invalid(room))
+                {
+                    MapCoord coord_x = subtile_coord_center(room->central_stl_x);
+                    MapCoord coord_y = subtile_coord_center(room->central_stl_y);
+                    event_create_event_or_update_nearby_existing_event(coord_x, coord_y,
+                        EvKind_RoomUnderAttack, room->owner, 0);
+                    if (is_my_player_number(room->owner))
+                    {
+                        output_message(SMsg_EnemyDestroyRooms, MESSAGE_DELAY_FIGHT, true);
```

Comment:
I don't know what's wrong with the code, but when I claim neutral rooms with this feature I get a sound message that says my rooms are under attack.

Reply:
> 
> 
> I don't know what's wrong with the code, but when I claim neutral rooms with this feature I get a sound message that says my rooms are under attack.

Odd. That only happens for me after the room has been claimed.

Reply:
By enemies you mean? In any case, the sounds that play should be the same when you claim in first person as when an imp claims on your behalf.

Reply:
I noticed this is fixed.